File: frontend/src/api/region.ts
import request from './request'

export interface Region {
  id: number
  code: string
  name: string
  mqtt_broker_url: string | null
  payment_endpoint: string | null
  ai_endpoint: string | null
  base_url: string | null
  status: string
  config: Record<string, any> | null
  description: string | null
  created_at: string
}

export interface RegionHealth {
  region: Region
  metrics: Record<string, number>
  overall_score: number
}

export interface GlobalMapItem {
  code: string
  name: string
  status: string
  overall_score: number
  metrics: Record<string, number>
}

export function getRegions() {
  return request.get('/regions') as Promise<Region[]>
}

export function createRegion(data: Partial<Region>) {
  return request.post('/regions', data) as Promise<Region>
}

export function updateRegion(id: number, data: Partial<Region>) {
  return request.put(`/regions/${id}`, data) as Promise<Region>
}

export function getRegionHealth(id: number) {
  return request.get(`/regions/${id}/health`) as Promise<RegionHealth>
}

export function getGlobalQualityMap() {
  return request.get('/regions/global-map') as Promise<GlobalMapItem[]>
}

File: backend/app/schemas/region.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RegionBase(BaseModel):
    code: str
    name: str
    mqtt_broker_url: str | None = None
    payment_endpoint: str | None = None
    ai_endpoint: str | None = None
    base_url: str | None = None
    config: dict | None = None
    description: str | None = None


class RegionCreate(RegionBase):
    pass


class RegionUpdate(BaseModel):
    name: str | None = None
    mqtt_broker_url: str | None = None
    payment_endpoint: str | None = None
    ai_endpoint: str | None = None
    base_url: str | None = None
    status: str | None = None
    config: dict | None = None
    description: str | None = None


class RegionResponse(RegionBase):
    model_config = ConfigDict(from_attributes=True)
    id: int
    status: str
    created_at: datetime


class RegionMetricResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    region_code: str
    metric_name: str
    metric_value: float
    computed_at: datetime


class RegionHealthSummary(BaseModel):
    region: RegionResponse
    metrics: dict[str, float]
    overall_score: float
